src/sites/twitch-twilight/modules/directory/following.js

```diff
@@ -0,0 +1,577 @@
+					{value: 2, title: 'Always'}
+				]
+			},
+
+			changed: () => this.router.current.name === 'dir-category' && this.ChannelCard.forceUpdate()
+		});
+
+		this.settings.add('directory.following.hide-boxart', {
+			default: 0,
+
+			ui: {
+				path: 'Directory > Following >> Placeholder',
+				title: 'Hide Boxart',
+				description: 'Do not display boxart over a stream / video thumbnail.',
+
+				component: 'setting-select-box',
+
+				data: [
+					{value: 0, title: 'Never'},
+					{value: 1, title: 'On Hover'},
+					{value: 2, title: 'Always'}
+				]
+			},
+
+			changed: value => {
+				this.css_tweaks.toggleHide('boxart-hide', value === 2);
+				this.css_tweaks.toggleHide('boxart-hover', value === 1);
+				this.router.current.name === 'dir-category' && this.ChannelCard.forceUpdate()
+			}
+		});
+
+		this.settings.add('directory.following.show-channel-avatar', {
+			default: 1,
+
+			ui: {
+				path: 'Directory > Following >> Placeholder',
+				title: 'Show Channel Avatar',
+				description: 'Show channel avatar next or on-top of a stream / video thumbnail.',
+
+				component: 'setting-select-box',
+
+				data: [
+					{value: 0, title: 'Never'},
+					{value: 1, title: 'Next to Stream Name'},
+					{value: 2, title: 'On Thumbnail, Hidden on Hover'},
+					{value: 3, title: 'On Thumbnail'}
+				]
+			},
+
+			changed: value => {
+				this.css_tweaks.toggleHide('profile-hover-following', value === 2);
+				this.router.current.name === 'dir-category' && this.ChannelCard.forceUpdate();
+			}
+		});
+
+		this.apollo.registerModifier('FollowedIndex_CurrentUser', `query {
+			currentUser {
+				followedLiveUsers {
+					nodes {
+						profileImageURL(width: 70)
+						stream {
+							createdAt
+						}
+					}
+				}
+				followedHosts {
+					nodes {
+						profileImageURL(width: 70)
+						hosting {
+							profileImageURL(width: 70)
+							stream {
+								createdAt
+								type
+							}
+						}
+					}
+				}
+			}
+		}`);
+
+		this.apollo.registerModifier('FollowingLive_CurrentUser', `query {
+			currentUser {
+				followedLiveUsers {
+					nodes {
+						profileImageURL(width: 70)
+						stream {
+							createdAt
+						}
+					}
+				}
+			}
+		}`);
+		
+		this.apollo.registerModifier('FollowingHosts_CurrentUser', `query {
+			currentUser {
+				followedHosts {
+					nodes {
+						profileImageURL(width: 70)
+						hosting {
+							profileImageURL(width: 70)
+							stream {
+								createdAt
+							}
+						}
+					}
+				}
+			}
+		}`);
+
+		this.ChannelCard = this.fine.define(
+			'channel-card',
+			n => n.renderGameBoxArt && n.renderContentType
+		);
+
+		this.apollo.registerModifier('FollowedIndex_CurrentUser', res => {
+			if (this.router.current.name === 'dir-category') {
+				this.modifyLiveUsers(res);
+				this.modifyLiveHosts(res);
+			}
+		}, false);
+
+		this.apollo.registerModifier('FollowingLive_CurrentUser', res => this.router.current.name === 'dir-category' && this.modifyLiveUsers(res), false);
```

Comment:
these 4 calls shouldn't be conditioned on the router state, the query can be reused from a different page's results and end up without the data